src/Confluent.Kafka.Avro/GenericSerializerImpl.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright 2016-2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using Confluent.SchemaRegistry;
+using Avro.Generic;
+using Avro.IO;
+
+
+namespace Confluent.Kafka.Serialization
+{
+    internal class GenericSerializerImpl : IAvroSerializerImpl<GenericRecord>
+    {
+        private ISchemaRegistryClient schemaRegistryClient;
+        private bool autoRegisterSchema;
+        private int initialBufferSize;
+        private bool isKey;
+
```

Comment:
This class seems to lack locking?

Reply:
yes, good point. in the serializer case, locking would seem very important... using thread safe collections will suffice.